### ISSUE TITLE
Ability to set defaults

### ISSUE THREAD

**Issue by sindresorhus** [jfmengels/eslint-ava-rule-tester#1]
https://github.com/sindresorhus/eslint-plugin-xo/pull/45#discussion_r73125085

I don't see a problem with this as long as it's clearly documented. The original RuleTester isn't very well documented anyways. We should improve upon it where it makes sense.

**Comment by jfmengels** [comment]
> I don't see a problem with this as long as it's clearly documented

True, but the documentation would be in a different project, so not "easily" accessible?

How about exporting a `createTestCase` (name TBD) function as part of this tool (or new module) that creates a test case with defaults?

```js
const testCase = createTestCase({
  parserOptions: { ... }
});

const errorTestCase = createTestCase({
  parserOptions: { ... },
  errors: [{
    ruleId: 'explicit-length-check',
    message: '`length` property should be compared to a value'
  }]
});

ruleTester.run('explicit-length-check', rule, {
  valid: [
    testCase('array.foo'),
    testCase('array.length')
  ],
  invalid: [
    errorTestCase('if ([].length) {}'), // No other options, just pass a string
    errorTestCase({
      code: 'if ([].length) {}',
      parserOptions: {} // other parserOptions
    }),
    errorTestCase({
      code: 'if ([].length) {}',
      errors: [...] // other errors
    })
  ]
});
```

Something like
```js
function createTestCase(defaultOptions) {
  return function(options) {
    if (typeof options === 'string') {
      return Object.assign({}, defaultOptions, {code: options});
    }
    return Object.assign({}, defaultOptions, options);
  };
}
```
I like the explicitness of this.

**Comment by sindresorhus** [comment]
> True, but the documentation would be in a different project, so not "easily" accessible?

It would actually be more accessible. People would go here to know about *this* module, not the ESLint docs. This module is already different from RuleTester in that it's for AVA. And it's not like RuleTester is very well documented.

> How about exporting a createTestCase (name TBD) function as part of this tool (or new module) that creates a test case with defaults?

I see a use-case for both. For the common case, your suggestion is more verbose. Having an option called `testCaseDefaults` (or something) would be much more concise. The whole point of RuleTester is to reduce redundant boilerplate.

**Comment by jfmengels** [comment]
Let's do this. I'm not sure when I'd get to this, so if someone wants to have a try at it, please let me know here :)
